platform-modules/manager/src/main/java/org/ekstep/taxonomy/mgr/impl/ContentManagerImpl.java

```diff
@@ -2356,4 +2359,260 @@ public Response syncHierarchy(String identifier) {
+			List<String> generatedDialCodes = (List<String>)result.get(ContentAPIParams.dialcodes.name());
+			if(!generatedDialCodes.isEmpty())
+				return generatedDialCodes;
+			else
+				throw new ServerException(TaxonomyErrorCodes.SYSTEM_ERROR.name(),
+						"Something Went Wrong While Processing Your Request. Please Try Again After Sometime!");
+		}else {
+			if (generateResponse.getResponseCode() == ResponseCode.CLIENT_ERROR)
+				throw new ClientException(generateResponse.getParams().getErr(), generateResponse.getParams().getErrmsg());
+			else throw new ServerException(TaxonomyErrorCodes.SYSTEM_ERROR.name(),
+					"Something Went Wrong While Processing Your Request. Please Try Again After Sometime!");
+		}
+		
+	}
+
+	private Optional<List<String>> getList(String[] params) {
+		return Optional.ofNullable(params).map(dialcodes -> new ArrayList<>(Arrays.asList(params)));
+	}
+
+	private Optional<List<String>> getReservedDialCodes(Node node) {
+		return getList((String[]) node.getMetadata().get(ContentAPIParams.reservedDialcodes.name()));
+	}
+
+	private List<String> validateAndGetReservedDialCodes(Node node) {
+		return getReservedDialCodes(node).
+				orElseThrow(() ->
+					new ClientException(ContentErrorCodes.ERR_NO_RESERVED_DIALCODES.name(),
+						"Error! No Dialcodes are Reserved for content:: " + node.getIdentifier()));
+	}
+
+	private boolean isContent(Node node) {
+		return StringUtils.equalsIgnoreCase(ContentAPIParams.Content.name(), node.getObjectType());
+	}
+
+	private void validateIsContent(Node node) {
+		if (!isContent(node))
+			throw new ClientException(ContentErrorCodes.ERR_NOT_A_CONTENT.name(), "Error! Not a Content.");
+	}
+
+	private boolean isRetired(Map<String, Object> metadata) {
+		return StringUtils.equalsIgnoreCase((String) metadata.get(ContentAPIParams.status.name()), ContentAPIParams.Retired.name());
+	}
+
+	private void validateIsNodeRetired(Map<String, Object> metadata) {
+		if (isRetired(metadata))
+			throw new ResourceNotFoundException(ContentErrorCodes.ERR_CONTENT_NOT_FOUND.name(), "Error! Content not found with id: " + metadata.get("identifier"));
+	}
+
+	/**
+	 * Implementation for {@link IContentManager#releaseDialcodes(String, String)}
+	 *
+	 * @param contentId
+	 * @param channelId
+	 * @return
+	 * @throws Exception
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public Response releaseDialcodes(String contentId, String channelId) throws Exception {
+        if(StringUtils.isBlank(channelId))
+            throw new ClientException(ContentErrorCodes.ERR_CHANNEL_BLANK_OBJECT.name(),
+                    "Channel can not be blank.");
+
+	    if (StringUtils.isBlank(contentId))
+			throw new ClientException(ContentErrorCodes.ERR_CONTENT_BLANK_OBJECT.name(),
+					"Content Id Can Not be blank.");
+
+		Node node = getContentNode(TAXONOMY_ID, contentId, null);
+        Map<String, Object> metadata = node.getMetadata();
+
+		validateIsContent(node);
```

Comment:
There is no proper formatting of the code.

Reply:
Refactored.